lib/src/main/java/com/hootsuite/nachos/NachoTextView.java

```diff
@@ -727,6 +727,128 @@ public void setTextWithChips(@Nullable List<ChipInfo> chips) {
     }
 
+    /**
+     *
+     * @return String of unchipped text that AutoCompleteTextView sees
+     */
+
+    public String getTextForSuggestion() {
+        if (getChipTokenizer() != null) {
+            int end = getSelectionEnd();
+            Editable editable = getText();
+            int start = getChipTokenizer().findTokenStart(editable, end);
+            if (start != -1 && end != -1) {
+                if (end < start) {
+                    start = end;
+                }
+                return editable.subSequence(start, end).toString();
+            }
+        }
+        return "";
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @return true if an existing chip contains text and null object
+     */
+    public boolean isAlreadyAdded(CharSequence text) {
+        return isAlreadyAdded(text, null);
+    }
+
+    /**
+     *
+     * @param text to check if exists with Chip
+     * @param object to check if exists with Chip
+     * @return true if an existing chip contains text and object
+     */
+    public boolean isAlreadyAdded(CharSequence text, Object object) {
+        List<Chip> chips = getAllChips();
+        for (Chip chip: chips) {
+            CharSequence existingText = chip.getText();
+            Object existingObject = chip.getData();
+            if (text.equals(existingText) &&
+                    (object == null && existingObject == null) || (object != null && object.equals(existingObject))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param text creates a chip with specified text
+     */
+    public void insertChip(CharSequence text) {
+        insertChip(text, null);
+    }
+
+    /**
+     *
+     * @param text creates a chip with specified text
+     * @param object creates a chip with specified object
+     */
+    public void insertChip(CharSequence text, Object object) {
+        if (mChipTokenizer != null) {
+            clearComposingText();
+
+            int end = getSelectionEnd();
+            Editable editable = getText();
+            int start = mChipTokenizer.findTokenStart(editable, end);
+            if (end != -1 && start != -1) {
+                if (end < start) {
+                    start = end;
+                }
+                editable.replace(start, end, mChipTokenizer.terminateToken(text, object));
+            } else {
+                CharSequence chippedText = mChipTokenizer.terminateToken(text, object);
+                getText().append(chippedText);
+            }
+        }
+    }
+
+    /**
+     *
+     * @param text remove chip if exists with text and null object
+     */
+    public void removeChip(CharSequence text) {
+        removeChip(text, null);
+    }
+
+    /**
+     *
+     * @param text remove if chip exists with text AND
+     * @param object remove if chip exits with object
+     */
+    public void removeChip(CharSequence text, Object object) {
```

Comment:
Removing a chip from the NachoTextView makes sense, however, since the current `NachoTextView` allows for non-unique chips (meaning duplicate chip values), this method will remove all the chips, maybe that is what you want. If this is the case, then you should definitely make that obvious in the javadoc. 

Maybe removeAllChips(String text) is better for the name.